src/libraries/System.Formats.Tar/tests/TarTestsBase.cs

```diff
@@ -611,5 +612,22 @@ public static IEnumerable<object[]> GetPaxAndGnuTestCaseNames()
                 yield return new object[] { name };
             }
         }
+
+        public static void ExtractWithTarTool(string source, string destination)
+        {
+            Assert.True(!OperatingSystem.IsWindows(), "This method is unsupported on Windows.");
+
+            using Process tarToolProcess = new Process();
+            tarToolProcess.StartInfo.FileName = "/usr/bin/tar";
+            tarToolProcess.StartInfo.Arguments = $"-xf {source} -C {destination}";
```

Comment:
We could add an additional boolean parameter `isCompressed` to this `ExtractWithTarTool` method. If `true`, then the arguments could include the `z` letter, which is used to extract `*.tar.gz` files.

We would just have to adjust line 271 of the `ResultingArchive_CanBeExtractedByOtherTools` test method, so that it takes a variable `CompressionMethod`.

Reply:
Yes, that should be part of the TODOs about adding more variants of this test.